src/server/floco.py

```diff
@@ -0,0 +1,213 @@
+from collections import OrderedDict
+from argparse import ArgumentParser
+from copy import deepcopy
+
+import numpy as np
+import torch
+from omegaconf import DictConfig
+from sklearn import decomposition
+
+from src.client.floco import FlocoClient
+from src.server.fedavg import FedAvgServer
+from src.utils.constants import NUM_CLASSES
+from src.utils.models import MODELS, DecoupledModel
+from src.utils.tools import Namespace
+
+
+class FlocoServer(FedAvgServer):
+
+    @staticmethod
+    def get_hyperparams(args_list=None) -> Namespace:
+        parser = ArgumentParser()
+        parser.add_argument("--endpoints", type=int, default=1)
+        parser.add_argument("--tau", type=int, default=100)
+        parser.add_argument("--rho", type=float, default=0.1)
+
+        # Floco+ (only used if pers_epoch > 0)
+        parser.add_argument("--pers_epoch", type=int, default=0)
+        parser.add_argument("--lamda", type=float, default=1)
+
+        return parser.parse_args(args_list)
+
+    def __init__(
+        self,
+        args: DictConfig,
+        algorithm_name: str = "Floco",
+        unique_model=False,
+        use_fedavg_client_cls=True,
+        return_diff=False,
+    ):
+        super().__init__(
+            args, algorithm_name, unique_model, use_fedavg_client_cls, return_diff
+        )
+        self.model = SimplexModel(self.args)
+        self.model.check_and_preprocess(self.args)
+        _init_global_params, _init_global_params_name = [], []
+        for key, param in self.model.named_parameters():
+            _init_global_params.append(param.data.clone())
+            _init_global_params_name.append(key)
+        self.public_model_param_names = _init_global_params_name
+        self.public_model_params: OrderedDict[str, torch.Tensor] = OrderedDict(
+            zip(_init_global_params_name, _init_global_params)
+        )
```

Comment:
I'm not 100% sure if there is a cleaner version to do this. We require the `SimplexModel` to be available at clients but could not find a way to initialize the `self.public_model_param_names` and `self.public_model_params` fields without copy-pasting the respective section from `FedAvgServer`.

Reply:
I will refactor this part if I find a cleaner way to do so.